lib/sandbox.js

```diff
@@ -416,6 +416,28 @@ return ((vm, host) => {
 			throw new VMError(`Failed to load '${moduleName}': Unknown type.`, 'ELOADFAIL');
 		};
 
+		_require.resolve = (request, options) => {
+			const resolveOption = vm.options.resolve;
+
+			if (!resolveOption) throw new VMError(`Access denied to resolve '${request}'`, 'EDENIED');
+			if (request == null) throw new VMError("Module '' not found.", 'ENOTFOUND');
+			if (typeof request !== 'string') throw new VMError(`Invalid module name '${request}'`, 'EINVALIDNAME');
```

Comment:
Is this the role of vm2 to check the value/type of the function arguments? I based these lines on what was written for the require function but I'm not sure if this is something you want here.